pkg/rpc/server/server.go

```diff
@@ -33,6 +34,13 @@ func NewScanServer(s scanner.Driver) *ScanServer {
 	return &ScanServer{localScanner: s}
 }
 
+// Log warning message on server side and return an error
+func WarnAndError(format string, args ...interface{}) error {
```

Comment:
nit: I'd unexport this function for now. And the function name would be `ErrorAndError` if we change warn to error😄 What about `teeError` or something like that?

```suggestion
func teeError(format string, args ...interface{}) error {
```

Reply:
Agree with you. `tee` is a more proper naming.